TwoWayDatabinding/app/src/main/java/com/enpassio/twowaydatabinding/data/ToyEntry.java

```diff
@@ -3,16 +3,13 @@
 import android.arch.persistence.room.Entity;
 import android.arch.persistence.room.Ignore;
 import android.arch.persistence.room.PrimaryKey;
-import android.databinding.BaseObservable;
-import android.databinding.Bindable;
-import android.util.Log;
-
-import com.enpassio.twowaydatabinding.BR;
+import android.support.annotation.Nullable;
 
+import java.util.HashMap;
 import java.util.Map;
 
 @Entity(tableName = "toys")
-public class ToyEntry extends BaseObservable {
+public class ToyEntry{
```

Comment:
I think this change is a significant improvement :+1:

Reply:
well, in fact it is because I recognized that it was not needed in our case. In some cases it would be needed. I thought of mentioning it in the article.

Reply:
@rajtheinnovator for instance, if we added a textview which would simulatenously show what we write to edittext, then we would need this(to notify simulatenously changes) . But in our sample we don't need it.

Reply:
Agree wit you here, @OyaCanli . However I've seen that some developers would communicate with database only when the operation is completed, that means when editing is done, else they'd just handle everything on UI using some local variables. I'm not very sure though that, how complex it could be when we're using it with ViewModels

Reply:
@rajtheinnovator But we as well, we don't communicate with the database until save is clicked. I created a temporary variable toyBeingModified which registers changes simulatenously. (It is kept in viewmodel) But it is not saved until user clicks save. 
And that wouldn't change if we add an extra textview or extend from BaseObservable. It doesn't notify database or observe database, if that is what yoy had thought. Its about UI. It notifies the widgets that use this piece of data.

Reply:
Yeah, that's what I meant @OyaCanli . You've covered that part well :clap: :raised_hands: